### ISSUE TITLE
Linux vs Windows mod mismatch

### ISSUE THREAD

**Issue by PasixTheValkyrie** [rwmt/Multiplayer#386]
Hello.

I've been trying to get a modded game working with my friend, who is playing on a Windows device, whilst I am playing on a Steam Deck.

I've tried several hours of getting to connect to his game, eventually sharing the entirety of his mod folder, config, etc needed to get it working.

I've noticed that the file sizes of the files he sends me change after I apply them and even though the files are exactly the same, I have a strong hunch that the file sizes are different because of newline encoding being different in Windows than it is in Linux.

The same config file for example was 8.7kb for him, whilst it was 8.4kb for me. I believe the \r\n vs \n difference might cause the Multiplayer mod to throw off a 'modified' file warning causing a myriad of issues.

**Comment by PasixTheValkyrie** [comment]
No, I can confirm this is happening consistently and you can very simply test it by having your Windows friends send over their config files. 

It will block the Linux player from connecting to the Windows user, but not the other way around.
Whether this is specifically a Proton thing or just Linux I can't tell for sure, however.

**Comment by notfood** [comment]
The files are untouched unless you open the settings. Rimworld can't edit them at all. Your files differ because they were touched. I basically never host as Linux. It's always the windows side.

**Comment by PasixTheValkyrie** [comment]
I'm afraid you're completely overlooking the issue as I wrote it.

It's not to do with Rimworld, it's most likely the newline encoding being
interpreted as a change between Linux (Proton?) and Windows.

The fact you mention you barely host as Linux kind of shows you're
overlooking what i'm raising as an issue.

Op vr 21 apr. 2023 05:57 schreef Meru ***@***.***>:

**Comment by notfood** [comment]
Even if you get the diff warning, you should be able to connect, the fact that isn't connecting means your configs differ on more than new lines. Approach support, this is not the right channel.

**Comment by Zetrith** [comment]
Line endings being significant is a valid issue. As for not being able to join, can you send a screenshot of the mismatch screen?

**Comment by PasixTheValkyrie** [comment]
These are the best I can offer you right now, as i'm not able to play with my pal right now.

![image](https://user-images.githubusercontent.com/43132310/233649012-a4574fdf-f25c-4a2e-8310-7d3486e5c3ac.png)
![image](https://user-images.githubusercontent.com/43132310/233649030-4198d994-f9dd-4bc4-b19e-dcb6c573759e.png)
![image](https://user-images.githubusercontent.com/43132310/233649046-3468ae67-d6e2-4822-8c9e-ebfd62392286.png)
![image](https://user-images.githubusercontent.com/43132310/233649053-57009091-cbfc-4dca-8506-61a82dd1f4ab.png)

Interesting to note is that all the files were exactly identical as I copied them directly from his files.

We were using Charlotte's Multiplayer-able modpack.

**Comment by Zetrith** [comment]
Regarding the mod files, is there a chance one side received a workshop update after you synchronized the files? Can you verify the game's file integrity in Steam?

**Comment by PasixTheValkyrie** [comment]
Can verify equal files. Did all the basics in the book, including verifying
game integrity. Only reason I made the github issue is since i've found no
other ways to fix it in several hours.

Op wo 26 apr. 2023 08:17 schreef Zetrith ***@***.***>:

**Comment by Zetrith** [comment]
If it isn't too much trouble, can you do a binary diff on the files in the mismatch screen?
